instrumentation/httpclient5/src/main/java/brave/httpclient5/HttpClient5Tracing.java

```diff
@@ -0,0 +1,90 @@
+
+  final HttpTracing httpTracing;
+
+  HttpClient5Tracing(HttpTracing httpTracing) {
+    this.httpTracing = httpTracing;
+  }
+
+  public static HttpClient5Tracing newBuilder(Tracing tracing) {
+    return new HttpClient5Tracing(HttpTracing.create(tracing));
+  }
+
+  public static HttpClient5Tracing newBuilder(HttpTracing httpTracing) {
+    return new HttpClient5Tracing(httpTracing);
+  }
+
+  public CloseableHttpClient create(HttpClientBuilder httpClientBuilder) {
+    httpClientBuilder.addExecInterceptorBefore(ChainElement.MAIN_TRANSPORT.name(),
+      HandleSendHandler.class.getName(),
+      new HandleSendHandler(httpTracing));
+    httpClientBuilder.addExecInterceptorBefore(ChainElement.PROTOCOL.name(),
+      HandleReceiveHandler.class.getName(),
+      new HandleReceiveHandler(httpTracing));
+    return httpClientBuilder.build();
+  }
+
+  public CloseableHttpAsyncClient create(HttpAsyncClientBuilder httpAsyncClientBuilder) {
+    final CurrentTraceContext currentTraceContext = httpTracing.tracing().currentTraceContext();
+    httpAsyncClientBuilder.addExecInterceptorBefore(PROTOCOL.name(),
+      AsyncHandleSendHandler.class.getName(),
+      new AsyncHandleSendHandler(httpTracing));
+    httpAsyncClientBuilder.addRequestInterceptorFirst(
+      (httpRequest, entityDetails, httpContext) -> HttpClientUtils.openScope(httpContext,
+        currentTraceContext));
+    httpAsyncClientBuilder.addRequestInterceptorLast(
+      (httpRequest, entityDetails, httpContext) -> HttpClientUtils.closeScope(httpContext));
+    httpAsyncClientBuilder.addResponseInterceptorFirst(
+      (httpResponse, entityDetails, httpContext) -> HttpClientUtils.openScope(httpContext,
+        currentTraceContext));
+    httpAsyncClientBuilder.addResponseInterceptorLast(
+      (httpResponse, entityDetails, httpContext) -> HttpClientUtils.closeScope(httpContext));
+    return new TracingHttpAsyncClient(httpAsyncClientBuilder.build(), currentTraceContext);
+  }
+
+  public CloseableHttpAsyncClient create(H2AsyncClientBuilder h2AsyncClientBuilder) {
+    final CurrentTraceContext currentTraceContext = httpTracing.tracing().currentTraceContext();
+    h2AsyncClientBuilder.addExecInterceptorBefore(PROTOCOL.name(),
+      AsyncHandleSendHandler.class.getName(),
+      new AsyncHandleSendHandler(httpTracing));
+    h2AsyncClientBuilder.addRequestInterceptorFirst(
+      (httpRequest, entityDetails, httpContext) -> HttpClientUtils.openScope(httpContext,
+        currentTraceContext));
+    h2AsyncClientBuilder.addRequestInterceptorLast(
+      (httpRequest, entityDetails, httpContext) -> HttpClientUtils.closeScope(httpContext));
+    h2AsyncClientBuilder.addResponseInterceptorFirst(
+      (httpResponse, entityDetails, httpContext) -> HttpClientUtils.openScope(httpContext,
+        currentTraceContext));
+    h2AsyncClientBuilder.addResponseInterceptorLast(
+      (httpResponse, entityDetails, httpContext) -> HttpClientUtils.closeScope(httpContext));
```

Comment:
instead of this, make HttpClientUtils.closeScope/openScope a singleton. This also allows you to give the class a name which can be helpful in troubleshooting vs a lambda

I don't know the function addResponseInterceptorLast accepts, but say..

```java
// top-level file, but package private
final class TraceContextInterceptor implements RequestInterceptor, ResponseInterceptor {
  final CurrentTraceContext currentTraceContext;
 --snip--
```

Then, when you make new instance of HttpClient5Tracing, you save one of these as a field, and use it everywhere. it makes the instantiation look cleaner also.